plugins/spam.py

```diff
@@ -3,32 +3,25 @@
 from errbot import BotPlugin
 from errbot.templating import tenv
 
-from plugins import constants
+from utils.mixin import DefaultConfigMixin
 
 
-class SpammingAlert(BotPlugin):
+class SpammingAlert(DefaultConfigMixin, BotPlugin):
     """
     A plugin which alerts the user that they might be spamming.
     """
 
-    CONFIG_TEMPLATE = {
```

Comment:
what the??

a plugin should include its own config template in its own source file, and not *rely* on config.py.

the plugin config information stays here in the plugin.

`config.py` should only have *overrides* for the orgs preferred starting config, only where it differs from the bot defaults.

otherwise it is mandatory to include in `config.py` a bunch of plugin config values which are not coala specific.

and if someone copies only the plugin file , they should be ok.  (they will also need to copy `DefaultConfigMixin` , which is annoying but we can solve that later.)  `constants.py` should go away eventually when each plugin is able to be customised by orgs easily.